feat: add examples mapping object (#95)
open-rpc/spec: README.md

<a name="method-errors"></a>errors | [[Error Object](#error-object) \| [Reference Object](#reference-object)] | A list of custom application defined errors that MAY be returned. The Errors MUST have unique error codes.
<a name="method-links"></a>links | [[Link Object](#link-object) \| [Reference Object](#reference-object)] | A list of possible links from this method call.
<a name="method-param-structure"></a>paramStructure | `"by-name"` | `"by-position"` | Format the server expects the params. Defaults to `"by-positon"`.

> <a name="method-example-mapping"></a>examples | [[Example Pairing Object](#example-pairing-object)] | Examples of the parameters and result pairing.

Reviewer: ```suggestion
<a name="method-examples"></a>examples | [[Example Pairing Object](#example-pairing-object)] | Array of [Example Pairing Object](#example-pairing-object) where each example includes a valid params-to-result [Content Descriptor](#content-descriptor) pairing.
```

> <a name="method-examples"></a>examples | [[Example Pairing Object](#example-pairing-object)] | Array of [Example Pairing Object](#example-pairing-object) where each example includes a valid params-to-result [Content Descriptor](#content-descriptor) pairing.

Author: ```suggestion
<a name="method-examples"></a>examples | [[Example Pairing Object](#example-pairing-object)] | Array of [Example Pairing Object](#example-pairing-object) where each example includes a valid params-to-result [Content Descriptor](#content-descriptor-Object) pairing.
```

}
```

###### Example Mapping Object

Field Name | Type | Description
---|:---:|---
<a name="example-params"></a>params | [[Example Object](#example-object)] | Example parameters.

> <a name="example-result"></a>result | [[Example Object](#example-object)] | Example results.

Author: ```suggestion
<a name="example-result"></a>result | [Example Object](#example-object) | Example results.
```

Reviewer: result*

This object MAY be extended with [Specification Extensions](#specification-extensions).

> Example Mapping Object Examples:

Reviewer: ![giphy 3](https://user-images.githubusercontent.com/1787231/53607764-073a8980-3b75-11e9-9c4e-5313d301cb5d.gif)

Author: `Example Example Examples:`

}
```

##### Example Pairing Object
The example Pairing object consists of a set of example params and result. The result is what you can expect from the JSON-RPC service given the exact params. 

Field Name | Type | Description
---|:---:|---

> <a name="example-params"></a>params | [[Example Object](#example-object)] | Example parameters.

Reviewer: ```suggestion
<a name="example-pairing-name"></a>name | `string` | Name for the example pairing.
<a name="example-pairing-params"></a>params | [[Example Object](#example-object)] | Example parameters.
```

> <a name="example-result"></a>result | [[Example Object](#example-object)] | Example results.

Reviewer: ```suggestion
<a name="example-pairing-result"></a>result | [[Example Object](#example-object)] | Example results.
```
